feat: DIA-1888: [FE] JWT auth token

nick-skriabin opened the pull request:
### PR fulfills these requirements
- [ ] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [ ] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [ ] Frontend



### Describe the reason for change
_(link to issue, supportive screenshots etc.)_



#### What does this fix?
_(if this is a bug fix)_



#### What is the new behavior?
_(if this is a breaking or feature change)_



#### What is the current behavior?
_(if this is a breaking or feature change)_



#### What libraries were added/updated?
_(list all with version changes)_



#### Does this change affect performance?
_(if so describe the impacts positive or negative)_



#### Does this change affect security?
_(if so describe the impacts positive or negative)_



#### What alternative approaches were there?
_(briefly list any if applicable)_



#### What feature flags were used to cover this change?
_(briefly list any if applicable)_



### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit



### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

yyassi-heartex committed:
feat: OPTIC-1359: Account Settings page moved to React

yyassi-heartex committed:
more changes

yyassi-heartex committed:
adding basic structure

yyassi-heartex committed:
toggling news letter properly

yyassi-heartex committed:
more tweaks

yyassi-heartex committed:
extracting input into InputFile

yyassi-heartex committed:
adding cursor

yyassi-heartex committed:
InputFile behavior in keyboard is now good

yyassi-heartex committed:
user avatar is updated properly

yyassi-heartex committed:
making sure menubar userpic updates at the same time

yyassi-heartex committed:
personalInfo in a good spot

yyassi-heartex committed:
inprogress tweaks

yyassi-heartex committed:
preping core to work with pages

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1359/my-account

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1359/my-account

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
adding all needed fields to the token section

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1359/my-account

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1359/my-account

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1359/my-account

yyassi-heartex committed:
PAT and MI should be good

yyassi-heartex committed:
putting all the sections back to a working state

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1359/my-account

yyassi-heartex committed:
things appear to be functional but we need to unify a bunch of components

yyassi-heartex committed:
getting navigation to work

pakelley committed:
wip: add jwt auth

pakelley committed:
Merge branch 'develop' into 'fb-dia-1839'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13084943381

pakelley committed:
only use jwt for orgs with it enabled

pakelley committed:
add simplejwt dep

pakelley committed:
log when basic token auth is used

pakelley committed:
enable jwt token auth for newly created orgs

pakelley committed:
move basic jwt token to LSO

pakelley committed:
fix app naming

pakelley committed:
Merge branch 'develop' into fb-dia-1839

pakelley committed:
update lockfile hash

pakelley committed:
add missing serializer

pakelley committed:
fix migration

pakelley committed:
separate settings

pakelley committed:
move middleware

pakelley committed:
block old tokens

pakelley committed:
add tests

pakelley committed:
remove ttl_days

pakelley committed:
fix swagger docs

pakelley committed:
add tests

pakelley committed:
remove unnecessary fields

nick-skriabin committed:
Merge branch 'fb-optic-1359/my-account' into fb-dia-1888/user-account

pakelley committed:
jwt access token FF

pakelley committed:
use standard .authenticate call

nick-skriabin committed:
Finalizing feature parity with the old implementation

nick-skriabin committed:
Export more stuff

pakelley committed:
test: improve tests

pakelley committed:
combine settings modules

pakelley committed:
fallback to other auth if jwt fails

pakelley committed:
add ability to revoke token

pakelley committed:
allow admin to make jwt settings changes

pakelley committed:
typo

nick-skriabin committed:
JWT token (mocked) + FF

nick-skriabin committed:
Remove FF override

nick-skriabin committed:
Merge remote-tracking branch 'origin/fb-dia-1839' into fb-dia-1888/user-account

pakelley committed:
Merge branch 'develop' into fb-dia-1839

pakelley committed:
fix hash

pakelley committed:
improve serializer behavior

pakelley committed:
fix api docs

nick-skriabin committed:
Token stuff

nick-skriabin committed:
Merge branch 'fb-dia-1839' into fb-dia-1888/user-account

pakelley committed:
lint

pakelley committed:
lint

pakelley committed:
change log level to info

pakelley committed:
don't need hasattr

pakelley committed:
separate settings for legacy vs jwt api tokens

nick-skriabin committed:
Update

nick-skriabin committed:
Merge branch 'fb-dia-1839' into fb-dia-1888/user-account

nick-skriabin committed:
Migrating APIProxy to lib/core + some other required migrations

pakelley committed:
make tokens last effectively forever

nick-skriabin committed:
Merge branch 'develop' into fb-dia-1888/user-account

nick-skriabin committed:
Handle errors

pakelley committed:
better testing for auth enablement matrix

pakelley committed:
fix token settings for new orgs

pakelley committed:
improve testing

pakelley committed:
lint

pakelley committed:
improve exception handling

pakelley committed:
add jwt urls

pakelley committed:
add utils file

pakelley committed:
fix imports

pakelley committed:
lint

nick-skriabin committed:
Merge branch 'fb-dia-1839' into fb-dia-1888/user-account

nick-skriabin committed:
Fix styling for LSE

pakelley committed:
improve exception when user does not exist

pakelley committed:
update urls

pakelley committed:
lint

pakelley committed:
lint

pakelley committed:
improve token list behavior

nick-skriabin committed:
Merge branch 'fb-dia-1839' into fb-dia-1888/user-account

nick-skriabin committed:
Token deletion optimistic

pakelley committed:
only return refresh tokens

nick-skriabin committed:
Finalize token creation

nick-skriabin committed:
Merge branch 'fb-dia-1839' into fb-dia-1888/user-account

pakelley committed:
lint

nick-skriabin committed:
Merge branch 'fb-dia-1839' into fb-dia-1888/user-account

nick-skriabin committed:
Remove excess mappings

pakelley committed:
exclude blacklisted tokens from token list

nick-skriabin committed:
Merge branch 'fb-dia-1839' into fb-dia-1888/user-account

pakelley committed:
return 404 when blacklisting already-blacklisted token

pakelley committed:
include legacy token settings in settings serializer

pakelley committed:
add tests for blacklist statuses

pakelley committed:
lint

pakelley committed:
only enable new auth if FF turned on

pakelley committed:
doc: get_queryset behavior

pakelley committed:
test: fix test

pakelley committed:
don't 401 directly when JWT auth fails

nick-skriabin committed:
Token settings (WIP)

nick-skriabin committed:
Merge branch 'fb-dia-1839' into fb-dia-1888/user-account

nick-skriabin committed:
Merge branch 'develop' into fb-dia-1888/user-account

nick-skriabin committed:
Token Settings

nick-skriabin committed:
TTL number and settings debounce

nick-skriabin committed:
Fix lint issues

pakelley committed:
improve woding on legacy token auth phaseout

bmartel commented on web/libs/core/src/pages/AccountSettings/AccountSettings.tsx:
```diff
@@ -0,0 +1,72 @@
+import { Card, CardHeader, CardTitle, CardContent, CardDescription } from "@humansignal/ui/lib/card-new/card";
```
Can we clean up the imports to being something a bit more concise than the full folder path? Doesn't have to be right now, I just think going forward if the paths are all long it appears messy similar to relative imports.

nick-skriabin replied:
suggestions? the only thing i see is to add `@humansignal/components` alias that points directly to the `lib/` folder

pakelley committed:
one more wording change

bmartel commented on web/libs/core/src/pages/AccountSettings/AccountSettings.tsx:
```diff
@@ -0,0 +1,72 @@
+import { Redirect } from "react-router-dom";
+import styles from "./AccountSettings.module.scss";
+import { accountSettingsSections } from "./sections";
+import { SidebarMenu } from "/apps/labelstudio/src/components/SidebarMenu/SidebarMenu";
```
We shouldn't do this, it is a cyclic import across packages, importing an app based thing in the package that later uses this package in the app.

bmartel commented on web/libs/core/src/pages/AccountSettings/atoms/index.ts:
```diff
@@ -0,0 +1,61 @@
+      return result as Partial<AuthTokenSettings>;
+    },
+    async onMutate(settings: Partial<AuthTokenSettings>) {
+      await queryClient.cancelQueries({ queryKey: [TOKEN_SETTINGS_KEY] });
+      const previousSettings = queryClient.getQueryData([TOKEN_SETTINGS_KEY]) as AuthTokenSettings;
+      queryClient.setQueryData(
+        [TOKEN_SETTINGS_KEY],
+        (old: AuthTokenSettings[]) => ({ ...previousSettings, ...settings }) as AuthTokenSettings,
+      );
+
+      return { previousSettings };
```
Can you add a comment on what this is doing

pakelley committed:
tidying

pakelley committed:
tidying

nick-skriabin committed:
Merge branch 'fb-dia-1839' into fb-dia-1888/user-account

nick-skriabin committed:
Fixing minor UI issues

nick-skriabin committed:
Describe optimistic update mechanism

nick-skriabin committed:
Merge branch 'develop' into fb-dia-1888/user-account

nick-skriabin committed:
Sync develop

nick-skriabin committed:
Account page FF

pakelley commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13277212549)
> **Error**: Workflow failed

nick-skriabin committed:
Core components -> blocks

pakelley commented:
Sorry, just saw that you already got everything merged (I canceled the `/git merge develop`)

nick-skriabin committed:
Minor UI changes

- Add "Expires on ..." to the date to clarify what the timestamp is
- Fix cURL example host name

nick-skriabin committed:
Remove unnecessary code

nick-skriabin committed:
Remove `toClassName`

Redundand code

nick-skriabin committed:
Adding a bunch of warnings about `/apps/labelstudio` imports

nick-skriabin committed:
Remove excess

nick-skriabin committed:
Merge branch 'develop' into fb-dia-1888/user-account
